lib/mongo_client.js

```diff
@@ -285,14 +305,22 @@ Object.defineProperty(MongoClient.prototype, 'optionsValidationLevel', {
  * @return {Promise<MongoClient>} returns Promise if no callback passed
  */
 MongoClient.prototype.connect = function(callback) {
-  // Validate options object
-  const err = validOptions(this.s.options);
+  try {
```

Comment:
This is necessary to pass the validation error to the callback and ensures that the `should error on unsupported options` test passes.